[FSDP] Add the unittests for the _replace_by_prefix

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73553
* __->__ #73530

Improve the unittests for the utility functions

Differential Revision: [D34529335](https://our.internmc.facebook.com/intern/diff/D34529335/)

fegin committed:
[FSDP] Add the unittests for the _replace_by_prefix

Improve the unittests for the utility functions

Differential Revision: [D34529335](https://our.internmc.facebook.com/intern/diff/D34529335/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
Thank you for improving tests!

fegin committed:
Update on "[FSDP] Add the unittests for the _replace_by_prefix"


Improve the unittests for the utility functions

Differential Revision: [D34529335](https://our.internmc.facebook.com/intern/diff/D34529335/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add the unittests for the _replace_by_prefix"


Improve the unittests for the utility functions

Differential Revision: [D34529335](https://our.internmc.facebook.com/intern/diff/D34529335/)

[ghstack-poisoned]
